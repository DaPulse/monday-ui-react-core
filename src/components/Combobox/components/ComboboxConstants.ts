import { SubIcon } from "../../../types";

export const COMBOBOX_DIVIDER_ITEM = "combobox-divider";
export const COMBOBOX_CATEGORY_ITEM = "combobox-category";
export const COMBOBOX_OPTION_ITEM = "combobox-option";

export enum ComboboxOptionIconType {
  DEFAULT = "default",
  RENDERER = "renderer"
}

export type ComboboxCategoryMap = {
  [key: string]: ComboboxCategoryType;
};

export type ComboboxCategoryType = {
  label: string;
  id: string;
  ariaLabel?: string;
  onlyShowOnSearch?: boolean;
};

export type ComboboxOptionType = {
  id: string;
  categoryId: string;
  leftIcon: SubIcon | ((className: string) => JSX.Element);
  rightIcon: SubIcon | ((className: string) => JSX.Element);
  leftIconType: ComboboxOptionIconType;
  rightIconType: ComboboxOptionIconType;
  label: string;
  iconSize: number;
  disabled: boolean;
  selected: boolean;
  ariaLabel: string;
  belongToCategory: boolean;
  tooltipContent: string;
};

export type ComboboxItem = {
  height?: number;
  type?: string;
  category?: ComboboxCategoryType;
  categoryId?: string;
  id?: string;
  index?: number;
  withDivider?: boolean;
  className?: string;
  belongToCategory?: boolean;
  option?: ComboboxOptionType;
  optionRenderer?: (option: ComboboxOptionType) => JSX.Element;
  isActive?: boolean;
  optionLineHeight?: number;
  shouldScrollToSelectedItem?: boolean;
};

export type ComboboxOptionEvents = {
  onOptionClick: (
    event: MouseEvent | KeyboardEvent,
    index: number,
    option: ComboboxOptionType,
    mouseTriggered: boolean
  ) => void;
  onOptionLeave: (event: MouseEvent, index: number, option: ComboboxOptionType, mouseTriggered: boolean) => void;
  onOptionEnter: (event: MouseEvent, index: number, option: ComboboxOptionType, mouseTriggered: boolean) => void;
  onOptionHover?: (event: MouseEvent, index: number, option: ComboboxOptionType, mouseTriggered: boolean) => void;
};
