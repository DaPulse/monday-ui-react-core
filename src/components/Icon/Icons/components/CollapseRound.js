/* eslint-disable */
/* tslint:disable */
import PropTypes from 'prop-types';
import React from 'react';
const CollapseRound = ({size, ...props}) => (
  <svg viewBox="0 0 256 256" fill="currentColor" width={ size || "256" } height={ size || "256" } {...props}>
    <path d="M94.423 58.88C91.3475 55.1332 85.7382 54.7414 82.0613 58.0565 78.3917 61.3483 77.9918 67.0667 81.0232 70.9028L81.1026 71.0032 121.398 113.912C122.974 115.836 125.432 117 128.001 117 130.627 117 133.066 115.791 134.721 113.776L174.766 71.1563 174.842 71.0614C176.279 69.2602 177 67.1846 177 65.031 177 62.4068 175.93 59.8466 173.932 58.0523 170.201 54.6885 164.6 55.1929 161.444 59.0299L128.001 94.1403 94.423 58.88zM161.577 197.12C164.652 200.867 170.262 201.259 173.939 197.943 177.608 194.652 178.008 188.933 174.977 185.097L174.897 184.997 134.602 142.088C133.026 140.164 130.568 139 127.999 139 125.373 139 122.934 140.209 121.279 142.224L81.2337 184.844 81.158 184.939C79.7209 186.74 78.9999 188.815 78.9999 190.969 78.9999 193.593 80.0701 196.153 82.0683 197.948 85.7986 201.311 91.3997 200.807 94.5559 196.97L127.999 161.86 161.577 197.12z"
      fill="currentColor" />
    <path d="M256 128C256 198.692 198.692 256 128 256C57.3076 256 0 198.692 0 128C0 57.3076 57.3076 0 128 0C198.692 0 256 57.3076 256 128ZM240 128C240 189.856 189.856 240 128 240C66.1441 240 16 189.856 16 128C16 66.1441 66.1441 16 128 16C189.856 16 240 66.1441 240 128Z"
      fill="currentColor" fillRule="evenodd" clipRule="evenodd" />
  </svg>
);
CollapseRound.displayName = 'CollapseRound';
CollapseRound.propTypes = {
  size: PropTypes.string
}
export default CollapseRound;
/* tslint:enable */
/* eslint-enable */
