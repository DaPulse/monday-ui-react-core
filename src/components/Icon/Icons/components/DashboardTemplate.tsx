/* eslint-disable */
/* tslint:disable */
import * as React from 'react';
export interface DashboardTemplateProps extends React.SVGAttributes<SVGElement> {
size?: string | number;
}
const DashboardTemplate: React.FC<DashboardTemplateProps> = ({size, ...props}) => (
  <svg viewBox="0 0 20 20" fill="currentColor" width={ size || "20" } height={ size || "20" } {...props}>
    <path d="M4 2C2.89543 2 2 2.89543 2 4V16C2 17.1046 2.89543 18 4 18H13.0556L12.3889 16.5H4C3.72386 16.5 3.5 16.2761 3.5 16V4C3.5 3.72386 3.72386 3.5 4 3.5H16C16.2761 3.5 16.5 3.72386 16.5 4V8.57143L18 7.92857V4C18 2.89543 17.1046 2 16 2H4ZM13.25 7.5C13.25 7.91421 13.5858 8.25 14 8.25C14.4142 8.25 14.75 7.91421 14.75 7.5V5.75C14.75 5.33579 14.4142 5 14 5C13.5858 5 13.25 5.33579 13.25 5.75V7.5ZM9.25 11.5C9.25 11.9142 9.58579 12.25 10 12.25C10.4142 12.25 10.75 11.9142 10.75 11.5V8C10.75 7.58579 10.4142 7.25 10 7.25C9.58579 7.25 9.25 7.58579 9.25 8L9.25 11.5ZM6.25 15C5.83579 15 5.5 14.6642 5.5 14.25L5.5 10.5C5.5 10.0858 5.83579 9.75 6.25 9.75C6.66421 9.75 7 10.0858 7 10.5V14.25C7 14.6642 6.66421 15 6.25 15Z"
      fill="currentColor" fillRule="evenodd" clipRule="evenodd" />
    <path d="M12.3478 9.03616C12.1563 8.92589 11.9215 9.08304 11.9505 9.30209L12.3311 12.1827C12.3426 12.2697 12.3105 12.3568 12.2454 12.4155L10.0882 14.3622C9.92419 14.5102 10.0011 14.782 10.2184 14.8222L13.0756 15.3503C13.1619 15.3663 13.2348 15.4237 13.2706 15.5038L14.4553 18.1569C14.5454 18.3587 14.8277 18.3696 14.9331 18.1753L16.0143 16.1816L18.4925 19.6049C18.7354 19.9404 19.2043 20.0155 19.5398 19.7726C19.8753 19.5297 19.9504 19.0608 19.7075 18.7253L17.3008 15.4007L19.4139 15.1762C19.6336 15.1529 19.7312 14.8878 19.579 14.7276L17.5779 12.6208C17.5174 12.5572 17.4922 12.4679 17.5103 12.3821L18.1113 9.53929C18.157 9.32312 17.935 9.14838 17.7356 9.2436L15.1136 10.4958C15.0345 10.5336 14.9418 10.53 14.8657 10.4862L12.3478 9.03616Z"
      fill="currentColor" fillRule="evenodd" clipRule="evenodd" />
  </svg>  
);
DashboardTemplate.displayName = 'DashboardTemplate';
export default DashboardTemplate;
/* tslint:enable */
/* eslint-enable */
