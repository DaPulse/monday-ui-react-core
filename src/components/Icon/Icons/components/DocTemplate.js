/* eslint-disable */
/* tslint:disable */
import PropTypes from 'prop-types';
import React from 'react';
const DocTemplate = ({size, ...props}) => (
  <svg viewBox="0 0 20 20" fill="currentColor" width={ size || "20" } height={ size || "20" } {...props}>
    <path d="M16.4033 7.62725C16.4049 7.60733 16.4057 7.58719 16.4057 7.56686C16.4057 7.54653 16.4049 7.5264 16.4033 7.50648V7.483C16.4032 7.02626 16.2217 6.58807 15.8988 6.265L12.0884 2.45456C11.7653 2.1317 11.3271 1.95029 10.8704 1.9502H4.97333C4.51628 1.9502 4.07794 2.13176 3.75475 2.45495C3.43157 2.77814 3.25 3.21647 3.25 3.67353V16.3269C3.25 16.7839 3.43157 17.2223 3.75475 17.5454C4.07794 17.8686 4.51627 18.0502 4.97333 18.0502H9.54158C9.08358 17.6482 8.70539 17.1365 8.43823 16.5502H4.97333C4.9141 16.5502 4.8573 16.5267 4.81541 16.4848C4.77353 16.4429 4.75 16.3861 4.75 16.3269V3.67353C4.75 3.6143 4.77353 3.55749 4.81541 3.51561C4.8573 3.47372 4.9141 3.4502 4.97333 3.4502L10.0391 3.50488V6.59353C10.0391 7.05059 10.2206 7.48892 10.5438 7.81211C10.867 8.1353 11.3053 8.31686 11.7624 8.31686H14V8.31512H16.4033V7.62725ZM11.5391 4.02658L14.3293 6.81686H11.7624C11.7032 6.81686 11.6464 6.79333 11.6045 6.75145C11.5626 6.70956 11.5391 6.65276 11.5391 6.59353V4.02658ZM5.91895 10.6002C5.91895 10.1859 6.25473 9.85016 6.66895 9.85016H10.4764C10.8906 9.85016 11.2264 10.1859 11.2264 10.6002C11.2264 11.0144 10.8906 11.3502 10.4764 11.3502H6.66895C6.25473 11.3502 5.91895 11.0144 5.91895 10.6002ZM6.6691 13.0502C6.25489 13.0502 5.9191 13.3859 5.9191 13.8002C5.9191 14.2144 6.25489 14.5502 6.6691 14.5502H8.50098C8.91519 14.5502 9.25098 14.2144 9.25098 13.8002C9.25098 13.3859 8.91519 13.0502 8.50098 13.0502H6.6691ZM12.4629 9.91921C12.3019 9.8265 12.1045 9.95863 12.1289 10.1428L12.4489 12.5646C12.4585 12.6378 12.4316 12.711 12.3768 12.7604L10.5632 14.397C10.4253 14.5214 10.49 14.75 10.6726 14.7838L13.0748 15.2278C13.1474 15.2412 13.2087 15.2894 13.2387 15.3568L14.2348 17.5874C14.3105 17.757 14.5479 17.7662 14.6364 17.6029L15.5453 15.927L17.9058 19.1877C18.11 19.4698 18.5043 19.5329 18.7863 19.3287C19.0684 19.1245 19.1316 18.7303 18.9273 18.4482L16.6267 15.2702L18.4036 15.0814C18.5884 15.0618 18.6704 14.8389 18.5425 14.7042L16.86 12.933C16.8092 12.8795 16.788 12.8044 16.8033 12.7323L17.3086 10.3422C17.347 10.1605 17.1603 10.0136 16.9927 10.0936L14.7883 11.1464C14.7217 11.1781 14.6438 11.1751 14.5798 11.1383L12.4629 9.91921Z"
      fill="currentColor" fillRule="evenodd" clipRule="evenodd" />
  </svg>
);
DocTemplate.displayName = 'DocTemplate';
DocTemplate.propTypes = {
  size: PropTypes.string
}
export default DocTemplate;
/* tslint:enable */
/* eslint-enable */
