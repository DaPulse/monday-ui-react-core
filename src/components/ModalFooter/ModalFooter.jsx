import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import classes from "./ModalFooter.module.scss";

const ModalFooter = ({ className, children }) => {
  return <div className={cx(classes.container, className)}>{children}</div>;
};

ModalFooter.propTypes = {
  /**
   * class name to be add to the wrapper
   */
  className: PropTypes.string
};
ModalFooter.defaultProps = {
  className: ""
};

ModalFooter.displayName = "ModalFooter";

export default ModalFooter;
