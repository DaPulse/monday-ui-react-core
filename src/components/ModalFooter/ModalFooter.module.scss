@import "../../styles/themes.scss";

.mondayStyleModalFooter {
  margin-top: 24px;
}
