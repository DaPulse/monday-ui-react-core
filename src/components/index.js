export { default as Loader } from "./Loader/Loader";
export { default as SearchComponent } from "./Search/Search"; // TODO: remove when bumping to version 1.0.0
export { default as Search } from "./Search/Search";
export { default as InputField } from "./TextField/TextField"; // TODO: remove when bumping to version 1.0.0
export { default as TextField } from "./TextField/TextField";
export { default as Dialog } from "./Dialog/Dialog";
export { default as Tooltip } from "./Tooltip/Tooltip";
export { default as Button } from "./Button/Button";
export { default as FormattedNumber } from "./FormattedNumber/FormattedNumber";
export { default as LinearProgressBar } from "./ProgressBars/LinearProgressBar/LinearProgressBar";
export { default as Link } from "./Link/Link";
export { default as CustomSvgIcon } from "./Icon/CustomSvgIcon";

export { default as Checkbox } from "./Checkbox/Checkbox";