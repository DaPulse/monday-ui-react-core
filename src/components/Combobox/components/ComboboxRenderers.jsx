import React from "react";
import cx from "classnames";
import ComboboxOption from "components/Combobox/components/ComboboxOption/ComboboxOption";
import ComboboxCategory from "components/Combobox/components/ComboboxCategory/ComboboxCategory";
import Divider from "../../Divider/Divider";
import { COMBOBOX_DIVIDER_ITEM, COMBOBOX_CATEGORY_ITEM, COMBOBOX_OPTION_ITEM } from "./ComboboxConstants";
export function comboboxItemRenderer({ type, ...otherArgs }) {
  switch (type) {
    case COMBOBOX_DIVIDER_ITEM: {
      return dividerRenderer(otherArgs);
    }
    case COMBOBOX_CATEGORY_ITEM: {
      return categoryRenderer(otherArgs);
    }
    case COMBOBOX_OPTION_ITEM: {
      return optionRenderer(otherArgs);
    }
  }
}

export function dividerRenderer({ id }) {
  return <Divider className="combobox_category-divider" key={id} />;
}

export function categoryRenderer({ id, category, className }) {
  return <ComboboxCategory key={id} category={category} className={className} />;
}

export function optionRenderer({
  id,
  index,
  option,
  onOptionClick,
  onOptionEnter,
  onOptionLeave,
  optionLineHeight,
  optionRenderer,
  shouldScrollToSelectedItem,
  isActive,
  belongToCategory,
  isActiveByKeyboard
}) {
  return (
    <ComboboxOption
      index={index}
      key={id}
      option={option}
      optionRenderer={optionRenderer}
      isActive={isActive}
      isActiveByKeyboard={isActiveByKeyboard}
      onOptionClick={onOptionClick}
      onOptionHover={onOptionEnter}
      onOptionLeave={onOptionLeave}
      optionLineHeight={optionLineHeight}
      belongToCategory={belongToCategory}
      shouldScrollWhenActive={shouldScrollToSelectedItem}
    />
  );
}
