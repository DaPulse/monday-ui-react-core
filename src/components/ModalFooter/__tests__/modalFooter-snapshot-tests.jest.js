import React from "react";
import renderer from "react-test-renderer";
import ModalFooter from "../ModalFooter";
import { Button } from "components";

describe("ModalFooter renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer
      .create(
        <ModalFooter>
          <Button />
        </ModalFooter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
