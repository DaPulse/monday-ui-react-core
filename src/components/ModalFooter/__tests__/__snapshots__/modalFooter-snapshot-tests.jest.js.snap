// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFooter renders correctly with empty props 1`] = `
<div
  className="mondayStyleModalFooter"
>
  <button
    aria-disabled={false}
    className="monday-style-button monday-style-button--size-medium monday-style-button--kind-primary monday-style-button--color-primary"
    data-testid="button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  />
</div>
`;
